src/composables/airport.ts

```diff
@@ -168,6 +168,35 @@ export const getAircraftForAirport = (data: Ref<StoreOverlayAirport['data']>, fi
     return aircraft;
 };
 
+export const arrivalIntervals = (aircrafts: Ref<AirportPopupPilotList | null>, intervals: number, intervalLength: number) => {
+    const returnArray = computed<AirportPopupPilotStatus[][]>(() => {
+        const returnArray = [];
+        for (let i = 0; i < intervals; i++) {
+            const intervalArray: AirportPopupPilotStatus[] = [];
+            returnArray.push(intervalArray);
+        }
+
+        if (aircrafts.value?.arrivals) {
+            for (let i = 0; i < aircrafts.value?.arrivals.length; i++) {
+                const arrival = aircrafts.value?.arrivals[i];
+                if (!arrival.eta) continue;
+
+                const currentDate = new Date() as Date;
```

Comment:
Let's move this creation of function top so we don't create too many dates all the time